src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +714,49 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * The overrides the add_data function from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
+			$args = $this->get_normalized_script_args( $handle, $value );
+			if ( $args['in_footer'] ) {
+				parent::add_data( $handle, 'group', 1 );
+			}
+			return parent::add_data( $handle, $key, $args );
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Normalize the data inside $args parameter and support backward compatibility.
+	 *
+	 * @param string        $handle Name of the script.
+	 * @param array         $args     {
+	 *      Optional. Additional script arguments. Default empty array.
+	 *
+	 *      @type boolean   $in_footer    Optional. Default true.
+	 *      @type string    $strategy     Optional. Values blocking|defer|async .Default 'blocking'.
+	 * }
+	 * @return array        Normalized $args array.
+	 */
+	private function get_normalized_script_args( $handle, $args = array() ) {
+		$default_args = array(
+			'in_footer' => false,
+			'strategy'  => 'blocking',
+		);
+		// handle backward compatibility for $in_footer
+		if ( 'boolean' === gettype( $args ) && $args ) {
+			$args = array( 'in_footer' => true );
+		}
```

Comment:
We could do a strict value check here instead of first checking for a type. Alternately, you could use `is_bool` Instead of `gettype`, which is the more common practice for type checking values in WP. Also, be careful about starting all inline comments with an upper case and ending them in a full stop.

```suggestion
		// Handle backward compatibility for $in_footer.
		if ( 'true' ===  $args ) {
			$args = array( 'in_footer' => true );
		}
```